extensions/pdh/PerformanceDataMonitor.cpp

```diff
@@ -0,0 +1,283 @@
+    withDescription("Format of the created flowfiles")->
+    withAllowableValue<std::string>(JSON_FORMAT_STR)->
+    withAllowableValue(OPEN_TELEMETRY_FORMAT_STR)->
+    withDefaultValue(JSON_FORMAT_STR)->build());
+
+PerformanceDataMonitor::~PerformanceDataMonitor() {
+  if (pdh_query_ != nullptr)
+    PdhCloseQuery(pdh_query_);
+  for (PerformanceDataCounter* resource_consumption_counter : resource_consumption_counters_) {
+    delete resource_consumption_counter;
+  }
+}
+
+void PerformanceDataMonitor::onSchedule(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSessionFactory>& sessionFactory) {
+  setupMembersFromProperties(context);
+
+  std::vector<PerformanceDataCounter*> valid_counters;
+
+  for (PerformanceDataCounter* counter : resource_consumption_counters_) {
+    PDHCounterBase* pdh_counter = dynamic_cast<PDHCounterBase*> (counter);
+    if (pdh_counter != nullptr) {
+      if (pdh_query_ == nullptr)
+        PdhOpenQuery(NULL, NULL, &pdh_query_);
+      PDH_STATUS add_to_query_result = pdh_counter->addToQuery(pdh_query_);
+      if (add_to_query_result != ERROR_SUCCESS) {
+        logger_->log_error(("Error adding " + pdh_counter->getName() + " to query: " + std::to_string(add_to_query_result)).c_str());
+        delete counter;
+      } else {
+        valid_counters.push_back(counter);
+      }
+    } else {
+      valid_counters.push_back(counter);
+    }
+  }
+  resource_consumption_counters_ = valid_counters;
+  PdhCollectQueryData(pdh_query_);
+}
+
+void PerformanceDataMonitor::onTrigger(core::ProcessContext* context, core::ProcessSession* session) {
+  if (resource_consumption_counters_.empty()) {
+    logger_->log_error("No valid counters for PerformanceDataMonitor");
+    return;
+  }
+
+  std::shared_ptr<core::FlowFile> flowFile = session->create();
+  if (!flowFile) {
+    logger_->log_error("Failed to create flowfile!");
+    return;
+  }
+
+  if (pdh_query_ != nullptr)
+    PdhCollectQueryData(pdh_query_);
+
+  rapidjson::Document root = rapidjson::Document(rapidjson::kObjectType);
+  rapidjson::Value& body = prepareJSONBody(root);
+  for (PerformanceDataCounter* counter : resource_consumption_counters_) {
+    PDHCounterBase* pdh_counter = dynamic_cast<PDHCounterBase*>(counter);
+    if (pdh_counter != nullptr && pdh_counter->collectData() != ERROR_SUCCESS)
+      continue;
+    counter->addToJson(body, root.GetAllocator());
+  }
+  PerformanceDataMonitor::WriteCallback callback(std::move(root));
+  session->write(flowFile, &callback);
+  session->transfer(flowFile, Success);
+}
+
+void PerformanceDataMonitor::initialize(void) {
+  setSupportedProperties({ CustomPDHCounters, PredefinedGroups, OutputFormatProperty });
+  setSupportedRelationships({ PerformanceDataMonitor::Success });
+}
+
+rapidjson::Value& PerformanceDataMonitor::prepareJSONBody(rapidjson::Document& root) {
+  switch (output_format_) {
+    case (OutputFormat::kOpenTelemetry):
```

Comment:
I think the braces are unnecessary here.

Reply:
changed it in  [d5959c3](https://github.com/martinzink/nifi-minifi-cpp/commit/d5959c3b1b5d0f377e7b7f2c13cc22a484b2e4a6)